File: Core/Models/Events/ChatMessage.cs
using Core.Models.Serialization;
using Newtonsoft.Json;

namespace Core.Models.Events
{
    public class ChatMessage : BaseEvent
    {
        [JsonProperty("sender_steamid")]
        [JsonConverter(typeof(LongToStringConverter))]
        public long SenderSteamId { get; set; }

        [JsonProperty("sender_name")]
        public string SenderName { get; set; }

        [JsonProperty("sender_side")]
        public Side SenderSide { get; set; }

        [JsonProperty("is_sender_alive")]
        public bool IsSenderAlive { get; set; }

        [JsonProperty("text")]
        public string Text { get; set; }


        public ChatMessage(int tick, float seconds)
            : base(tick, seconds)
        {
        }
    }
}

File: src/common/types/chat-message.ts
import type { TeamNumber } from 'csdm/common/types/counter-strike';
import type { BaseEvent } from 'csdm/common/types/base-event';

export type ChatMessage = BaseEvent & {
  message: string;
  senderIsAlive: boolean;
  senderSide: TeamNumber;
  senderSteamId: string;
  senderName: string;
};
